profiling/utils.py

```diff
@@ -0,0 +1,364 @@
+import argparse
+import contextlib
+import functools
+import os
+import random
+import time
+from collections import defaultdict
+from itertools import chain
+
+import megatron.core.parallel_state as mpu
+import numpy as np
+import torch
+import torch.nn as nn
+import wandb
+from megatron.core.parallel_state import initialize_model_parallel
+from megatron.core.tensor_parallel import (
+    ColumnParallelLinear,
+    RowParallelLinear,
+    model_parallel_cuda_manual_seed,
+)
+from torch.profiler import profile
+
+from flex_model.core import FlexModel, HookFunction
+
+
+class TestNetwork(nn.Module):
```

Comment:
Also `TestNetwork` should be deprecated in the next testing PR in favour of Megatron-LM models for working with 3-D sharding schemes and training/finetuning workloads.